int-set/src/lib.rs

```diff
@@ -12,45 +12,92 @@
 //!
 //! When constructing a new IntSet from an existing lists of integer values the most efficient
 //! way to create the set is to initialize it from a sorted (ascending) iterator of the values.
+//!
+//! For a type to be stored in the IntSet it must implement the IntSetMember trait, and all
+//! unique values of that type must be able to be mapped to and from a unique u32 value.
+//! See the IntSetMember trait for more information.
 
 mod bitpage;
 mod bitset;
 
 use bitset::BitSet;
 use std::hash::Hash;
+use std::marker::PhantomData;
 use std::ops::RangeInclusive;
 
 /// A fast & efficient invertible ordered set for small (up to 32-bit) unsigned integer types.
 #[derive(Clone, Debug, Hash, PartialEq, Eq)]
-pub struct IntSet<T>(Membership<T>);
+pub struct IntSet<T>(Membership, PhantomData<T>);
+
+/// Members of IntSet must implement this trait. Members of IntSet's must meet the following
+/// conditions to be used in an IntSet:
+///
+/// 1. Every possible unique value of T must be able map to and from a unique u32
+///    integer.
+///
+/// 2. The mapped u32 values must retain the same ordering as the values in T.
+///
+/// 3. all_values_iter() must iterate over all values in T in sorted order (ascending).
```

Comment:
Nit: `.sorted()` that returns an iter would more obviously convey the nature of this fn

Reply:
if we're bikeshedding I might go with `ordered_values`.


Also an alternative way to implement this would be as a fn, `in_domain` that returns a bool?

Reply:
The iterator is probably better since you'll avoid having to process large unused parts of the u32 domain for smaller types like u16. ordered_values() sounds good to me, I'll change to that.